internal/controller/pkg/revision/hook.go

```diff
@@ -77,6 +80,18 @@ func (h *ProviderHooks) Pre(ctx context.Context, pkg runtime.Object, pr v1beta1.
 		return errors.New(errNotProvider)
 	}
 
+	provRev, ok := pr.(*v1beta1.ProviderRevision)
+	if !ok {
+		return errors.New(errNotProviderRevision)
+	}
+
+	if !cmp.Equal(provRev.Spec.PermissionRequests, pkgProvider.Spec.Controller.PermissionRequests) {
+		provRev.Spec.PermissionRequests = pkgProvider.Spec.Controller.PermissionRequests
+		if err := h.client.Update(ctx, provRev); err != nil {
+			return errors.Wrap(err, errUpdateProviderRevision)
+		}
+	}
```

Comment:
@hasheddan I don't love that I update the ProviderRevision post-create to propagate the requests rather than setting it at create time. Is there a better option here?

Reply:
We _could_ move pull logic to the package controller. That would be a pretty large change, and we would still need to access the package contents in the revision controller (this would be fine, the package controller and the revision controller would just share the cache), but it would allow the package manager to set revision fields that are specific to package contents at creation time. The original motivation for not doing this was just a separation of concern for the package manager to only worry about the lifecycle of revisions and for the revision itself to act in its own interest and have all "revision-specific" operations take place in its controller.

We could choose to annotate / label the service account we create with any additional permissions that we are requesting from the RBAC manager. Any thoughts on that?

Reply:
> We could move pull logic to the package controller.

I think I prefer this implementation to that one - I can convince myself we're "late intializing" the revision. :)

> We could choose to annotate / label the service account we create with any additional permissions that we are requesting from the RBAC manager. Any thoughts on that?

I prefer the revision to this alternative too - especially because it would be tricky to add RBAC rules as unstructured labels/annotations.

Reply:
> especially because it would be tricky to add RBAC rules as unstructured labels/annotations

Do you think marshaling the struct and adding it as an annotation, then unmarshaling in the RBAC manager would be error prone?

Reply:
It's not so much that I think it would be error prone so much as I don't see the advantage of doing so, especially when we can pass that data using a structured, schemafied object (i.e. the ProviderRevision's spec) instead.

Reply:
Gotcha 👍 I guess its more about that I feel like it doesn't belong in the spec, but given that it isn't intended to be a user-facing object then maybe it is fine.

Reply:
We set the requests in the status now, which feels like a better fit in this context.